### PR TITLE
SQL-2499: Add support for InfoType::SQL_KEYWORDS

### PR DESCRIPTION
**Pull request by mattChiaravalloti**
This PR adds support for `InfoType::SQL_KEYWORDS` by returning a `lazy_static` string representing MongoSQL-specific keywords. I took the `mongosql-rs` list of keywords ([here](https://github.com/10gen/mongosql-rs/blob/921ea1b6df6d2d9df360c12bcd49a46acddadacc/mongosql/src/parser/mongosql.lalrpop#L754)) and removed anything already included in the [ODBC Reserved Keywords list](https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/reserved-keywords?view=sql-server-ver16).

### CODE REVIEW

File: constants/src/lib.rs [hunk 1]
```diff
@@ -22,6 +22,39 @@ lazy_static! {
...
+        "JAVASCRIPTWITHSCOPE".to_string(),
+        "LIMIT".to_string(),
+        "LONG".to_string(),
+        "MAXKEY".to_string(),
+        "MINKEY".to_string(),
+        "MISSING".to_string(),
+        "NUMBER".to_string(),
+        "OBJECTID".to_string(),
+        "OFFSET".to_string(),
+        "ORDER".to_string(),
```
Comment (bucaojit) [inline constants/src/lib.rs:48]: I see `ORDER` in the ODBC reserved words, should we remove it here?

Comment (mattChiaravalloti) [reply constants/src/lib.rs:48]: Thanks, good catch!

**Review by bucaojit — APPROVED** [review]
LGTM

**Comment by mattChiaravalloti** [comment]
@bucaojit  @nbagnard since I'll be until the onsite, could one of you add this to the merge queue (or apply updates if needed) while I'm gone? No sense in having something this small wait over a week for me to get back! Thank you!

File: constants/src/lib.rs [hunk 2]
```diff
@@ -22,6 +22,38 @@ lazy_static! {
...
+    pub static ref KEYWORDS: String = [
+        "AGGREGATE".to_string(),
+        "ARRAY".to_string(),
+        "BINDATA".to_string(),
+        "BOOL".to_string(),
+        "BOOLEAN".to_string(),
+        "BSON_DATE".to_string(),
+        "BSON_TIMESTAMP".to_string(),
+        "DBPOINTER".to_string(),
+        "DEPTH".to_string(),
```
Comment (nbagnard) [inline constants/src/lib.rs:34]: I see [DEC](https://github.com/10gen/mongosql-rs/blob/921ea1b6df6d2d9df360c12bcd49a46acddadacc/mongosql/src/parser/mongosql.lalrpop#L780) in mongosql.

**Review by nbagnard — APPROVED** [review]
LGTM!

If you can add 'DEC', we will match mongosql keywords, but I don't expect applications to generate SQL with that keyword

File: constants/src/lib.rs (hunk 2 above)
Comment (mattChiaravalloti) [reply constants/src/lib.rs:34]: DEC is in the ODBC reserved keywords list already!

Comment (nbagnard) [reply constants/src/lib.rs:34]: You are right! I totally missed it!

**Pull request merged**